### PR TITLE
feat: verify-api

### PR DESCRIPTION
**Pull request by ganchoradkov**
# Description
`Verify API` implementation according to specs https://github.com/WalletConnect/walletconnect-docs/pull/436

## How Has This Been Tested?
tests
dogfooding

## Due Diligence

- [ ] Breaking change
- [x] Requires a documentation update
  - [] Related docs issue/PR (if docs are not included in this PR):
- [ ] Requires a e2e/integration test update

### CODE REVIEW

File: packages/core/src/controllers/verify.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    this.logger = generateChildLogger(logger, this.name);
+    this.verifyUrl = VERIFY_SERVER;
+    this.abortController = new AbortController();
+  }
+
+  public init: IVerify["init"] = async (params) => {
+    // ignore on non browser environments
+    if (!isBrowser() || !document) return;
+
+    this.verifyUrl = params?.verifyUrl || VERIFY_SERVER;
```
Comment (xzilja) [inline packages/core/src/controllers/verify.ts:28]: Since `this.verifyUrl` is set here with fallback, maybe safe to remove it from constructor?

**Review by xzilja — APPROVED** [review]
Looks good 👍 It seems like there are 2 approaches used to "abort" promises that are in flight, one is using abort controller in fetch and the other approach is a `Promise.race` for original promise and artificial delay one that rejects.

Abort one seems to be like a more correct approach here, so wanted to check if it can be used everywhere or are there any specifics here to why you used 2 approaches?

File: packages/core/src/controllers/verify.ts (hunk 1 above)
Comment (ganchoradkov) [reply packages/core/src/controllers/verify.ts:28]: it is, but typescript then complains about
- this.verifyUrl is not definitely assigned in the constructor
- requires everywhere to be checked that is assigned i.e.  `?.` on every usage

**Review by arein — APPROVED** [review]

**Review by pedrouid — APPROVED** [review]
👍

**Pull request merged**